exporters/deploytime/app.py

```diff
@@ -50,6 +55,34 @@ def collect(self) -> Iterable[GaugeMetricFamily]:
             )
             yield (metric)
 
+    def get_and_log_namespaces(self) -> set[str]:
+        """
+        Get the set of namespaces to watch, and log what they are.
+        They will be either:
+        1. The namespaces explicitly specified
+        2. The namespaces matched by PROD_LABEL
+        3. If the PROD_LABEL is given, that implicitly matches all namespaces.
+        """
+        if self._namespaces:
+            logging.info("Watching namespaces %s", self._namespaces)
+            return self._namespaces
+
+        logging.info(
+            "No namespaces specified, watching all namespaces with given PROD_LABEL (%s)",
+            self.prod_label,
+        )
+        all_namespaces = dyn_client.resources.get(api_version="v1", kind="Namespace")
+        namespaces = {
+            namespace.metadata.name
+            for namespace in all_namespaces.get(label_selector=prod_label).items
```

Comment:
@KevinMGranger are you saying that this will return all namespaces if `prod_label` is `null` or empty string? Even if that is the case, it seems like a very sideways way of defaulting to all namespaces rather then being explicite about it with an if statement that if no PROD_LABEL is supplied then use all namespaces. not to mention the confusion with th elogging mesage of `No namespaces specfiied, watching all namespaces with given PROD_LABEL (%s)` where PROD_LABEL is empty.

Reply:
You're right that the log message is confusing, I'll change that. But is this sideways? It explicitly retrieves all of them so that they can be logged.

Reply:
Ah, I see what you mean-- the behavior of querying for an empty string is confusing. I've addressed that in the latest commit, if you think that helps.